src/components/Form/Form.tsx

```diff
@@ -0,0 +1,46 @@
+import { Button } from 'components/Button/Button';
+import { FormField } from 'components/FormField/FormField';
+import { Title } from 'components/Title/Title';
+import { useState } from 'react';
+
+import styles from './Form.module.scss';
+import { FormProps, FormState } from './Form.types';
+
+export function Form({ title, fields, buttons, onSubmit, onChange }: FormProps) {
+  const [formData, setFormData] = useState<FormState>();
+
+  function handlerChange({ name, value }) {
+    setFormData((prevState) => {
+      return {
+        ...prevState,
+        [name]: value,
+      };
+    });
+
+    if (onChange) {
+      onChange(formData);
+    }
+  }
+
+  function handlerSubmit(event) {
+    event.preventDefault();
+    onSubmit(formData);
```

Comment:
🧐
Опционально, но работу с ивентом я бы тоже отбил
```suggestion
    event.preventDefault();
    
    onSubmit(formData);
```